pandas/io/common.py

```diff
@@ -1207,25 +1212,22 @@ def _maybe_memory_map(
 
     # need to open the file first
     if isinstance(handle, str):
-        if encoding and "b" not in mode:
-            # Encoding
-            handle = open(handle, mode, encoding=encoding, errors=errors, newline="")
-        else:
```

Comment:
any tests validate this? (nbd if you can't come up with an example, but ideally)

Reply:
This encoding+memory_map test should test it: https://github.com/pandas-dev/pandas/blob/962495d7c2d02acee2f76a26017362b9d08f497f/pandas/tests/io/parser/test_encoding.py#L234